### PR TITLE
Initial ADFS stuff - need to test on server to make more progress

### PR DESCRIPTION
**Pull request by ad-world**

### CODE REVIEW

File: server/auth/adfs.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import passport from "passport";
+import { IProfile, OIDCStrategy, VerifyCallback } from "passport-azure-ad";
+import express, { NextFunction, Request, Response } from "express";
+
+import "express-session";
+declare module "express-session" {
```
Comment (some-molasses) [inline server/auth/adfs.ts:6]: looks like type definitions might exist for this under `@types/express-session`?

File: server/auth/adfs.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      clientID: process.env.ADFS_CLIENT_ID ?? "",
+      responseType: "id_token",
+      responseMode: "form_post",
+      redirectUrl: process.env.REDIRECT_URI ?? "",
+      passReqToCallback: true,
+      useCookieInsteadOfSession: true,
+      cookieEncryptionKeys: [
+        {
+          key: process.env.COOKIE_ENCRYPTION_KEY ?? "",
+          iv: process.env.COOKIE_ENCRYPTION_IV ?? "",
```
Comment (some-molasses) [inline server/auth/adfs.ts:28]: Can you add examples of these new environment variables to the .env.example?

Also, should we add an error message if they are detected to not exist? (feel free to make this its own ticket to deal with after ADFS implementation is done)

**Review by some-molasses — APPROVED** [review]
looks good!  I think it may be valuable to put more error handling around the env variables but that can come later tbh.

File: server/auth/adfs.ts (hunk 1 above)
Comment (ad-world) [reply server/auth/adfs.ts:6]: `@types/express-session` is imported, but passport is a little bit outdated and is supposed to make this extension anyway. I'm planning on changing the `any` to something else once i figure out the type of this.

File: server/auth/adfs.ts (hunk 2 above)
Comment (ad-world) [reply server/auth/adfs.ts:28]: I'll add examples yes fs. Also what should desired behavior be if a var isn't detected? Log an error and exit gracefully? or should the app continue running

Comment (some-molasses) [reply server/auth/adfs.ts:28]: I think exit gracefully - most of these env variables are needed for the site to function securely, right?  We'd want the server admin to be very aware if something security-related is missing lol

**Review by some-molasses — APPROVED** [review]

**Pull request merged**